### PR TITLE
Feat/implement parse string

### PR DESCRIPTION
**Pull request by JoelVR17**
# Pull Request: Improve Function with Regex

## Description
This PR implements the `parse_string` function in the `verify_contract` and `verify_custom_api` routes to enhance the modularity of the task system, allowing for the use of dynamic strings when configuring tasks.

### Changes Made
- **Affected Files:**
  - `src/endpoints/quests/verify_contract.rs`
    - Implemented `parse_string` in `handler` to process the `call_data`.
    - Added regex validation to verify the contract call result.
  - `src/endpoints/quests/verify_custom_api.rs`
    - Applied `parse_string` to the `api_url` and improved API response handling.
  - `src/utils.rs`
    - Enhanced the `parse_string` function to utilize regex for pattern replacement in strings.

### Technical Details
- The `regex` library was used to handle patterns in strings, thereby improving validation and processing of dynamic inputs.
- Logic was added to convert addresses to hexadecimal and decimal using the `to_hex` function.

### CODE REVIEW

File: src/endpoints/quests/verify_custom_api.rs [hunk 1]
```diff
@@ -55,7 +56,9 @@ pub async fn handler(
     };
 
     // Call the specified API
-    let response = get(api_url).await;
+    let parsed_api_url = parse_string(api_url, FieldElement::from_str(&query.addr).unwrap());
+
+    let response = get(&parsed_api_url).await;
 
     match response {
         Ok(res) => {
```
Comment (Marchand-Nicolas) [inline src/endpoints/quests/verify_custom_api.rs:64]: Missing implementation of parse_string for the regex (you only did for the api_url)

File: src/endpoints/quests/verify_contract.rs [hunk 2]
```diff
@@ -79,14 +86,16 @@ pub async fn handler(
...
                 Ok(result) => {
                     let regex = match Regex::new(&call.regex) {
                         Ok(re) => re,
                         Err(e) => return get_error(format!("Invalid regex: {}", e)),
                     };
                     let result_str = result.iter().map(|&r| r.to_string()).collect::<Vec<String>>().join(",");
-                    
+
+        
                     if !regex.is_match(&result_str) {
```
Comment (Marchand-Nicolas) [inline src/endpoints/quests/verify_contract.rs:93]: Same here, parse_string should be implemented to the regex

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]
What you implemented is working really well, but you only implemented the parse_string function to call_data and and api_url, and forgot to do so with the regexs in both routes.  This would allow to have dynamic regex

File: src/endpoints/quests/verify_contract.rs (hunk 2 above)
Comment (JoelVR17) [reply src/endpoints/quests/verify_contract.rs:93]: I got it, I'll work on it

**Comment by JoelVR17** [comment]
@Marchand-Nicolas the changes has been realized. If you need another thing, please let me know

**Review by Marchand-Nicolas — APPROVED** [review]
Working really well! GG, lgtm

**Pull request merged**